### PR TITLE
Extension - Add validate account in case sign transaction with Ledger account

### PR DESCRIPTION
**Pull request by Thiendekaco**

### CODE REVIEW

File: packages/extension-koni-ui/src/hooks/ledger/useLedger.ts [hunk 1]
```diff
@@ -221,8 +221,19 @@ export function useLedger (slug?: string, active = true, isSigning = false, forc
       });
   }, [getLedger, ledger, t]);
 
-  const signTransaction = useCallback(async (message: Uint8Array, metadata: Uint8Array, accountOffset?: number, addressOffset?: number, accountOption?: Partial<AccountOptions>): Promise<LedgerSignature> => {
+  const signTransaction = useCallback(async (message: Uint8Array, metadata: Uint8Array, accountOffset?: number, addressOffset?: number, address?: string, accountOption?: Partial<AccountOptions>): Promise<LedgerSignature> => {
     if (ledger) {
+      const addressOnCurrentLedger = await ledger.getAddress(false, accountOffset, addressOffset);
```
Comment (S2kael) [inline packages/extension-koni-ui/src/hooks/ledger/useLedger.ts:226]: Add `accountOption`

File: packages/extension-koni-ui/src/hooks/ledger/useLedger.ts [hunk 2]
```diff
@@ -221,8 +221,19 @@ export function useLedger (slug?: string, active = true, isSigning = false, forc
...
+
+      if (addressOnCurrentLedger.address !== address) {
```
Comment (S2kael) [inline packages/extension-koni-ui/src/hooks/ledger/useLedger.ts:228]: Replace with deep compare(`isSameAddress`), `addressOnCurrentLedger` can be in network format

File: packages/extension-koni-ui/src/hooks/ledger/useLedger.ts [hunk 3]
```diff
@@ -242,8 +253,19 @@ export function useLedger (slug?: string, active = true, isSigning = false, forc
...
+        return new Promise((resolve, reject) => {
+          const error = new Error(t('Wrong device. Connect your previously used Ledger and try again'));
+
+          handleError(error);
+          reject(error);
+        });
+      }
+
```
Comment (S2kael) [inline packages/extension-koni-ui/src/hooks/ledger/useLedger.ts:268]: Same as above

**Review by S2kael — CHANGES_REQUESTED** [review]

**Review by S2kael — APPROVED** [review]

**Review by S2kael — APPROVED** [review]

**Comment by saltict** [comment]
🚀 Deployed on https://pr-3278--sw-web-runner.netlify.app

**Review by saltict — APPROVED** [review]

**Pull request merged**